barretenberg/cpp/src/barretenberg/proof_system/op_queue/ecc_op_queue.hpp

```diff
@@ -38,6 +38,69 @@ class ECCOpQueue {
 
     Point get_accumulator() { return accumulator; }
 
+    /**
+     * @brief Prepend the information from the previous queue (used before accumulation/merge proof to be able to run
+     * circuit construction separately)
+     *
+     * @param previous
+     */
+    void prepend_previous_queue(const ECCOpQueue& previous)
+    {
+        // Allocate enough space
+        std::vector<ECCVMOperation> raw_ops_updated(raw_ops.size() + previous.raw_ops.size());
+        // Copy raw_ops
+        std::copy(previous.raw_ops.begin(), previous.raw_ops.end(), raw_ops_updated.begin());
+        std::copy(raw_ops.begin(),
+                  raw_ops.end(),
+                  std::next(raw_ops_updated.begin(), static_cast<long>(previous.raw_ops.size())));
+
+        // Swap raw_ops underlying storage
+        raw_ops.swap(raw_ops_updated);
+        // Do the same 3 operations for ultra_ops
+        for (size_t i = 0; i < 4; i++) {
+            std::vector<Fr> current_ultra_op(ultra_ops[i].size() + previous.ultra_ops[i].size());
+            std::copy(previous.ultra_ops[i].begin(), previous.ultra_ops[i].end(), current_ultra_op.begin());
+            std::copy(ultra_ops[i].begin(),
+                      ultra_ops[i].end(),
+                      std::next(current_ultra_op.begin(), static_cast<long>(previous.ultra_ops[i].size())));
+            ultra_ops[i].swap(current_ultra_op);
+        }
+        // Update sizes
+        current_ultra_ops_size += previous.ultra_ops[0].size();
+        previous_ultra_ops_size += previous.ultra_ops[0].size();
+        // Update commitments
+        ultra_ops_commitments = previous.ultra_ops_commitments;
+        previous_ultra_ops_commitments = previous.previous_ultra_ops_commitments;
+    }
+
+    /**
+     * @brief Swap all internal values in this queue with the other
+     *
+     * @param other
+     */
+    void swap(ECCOpQueue* other)
```

Comment:
can't this be achieved with `std::swap`?

Reply:
Made it so you can use std::swap